src/anyio/_backends/_asyncio.py

```diff
@@ -2746,25 +2748,29 @@ async def wait_readable(cls, obj: HasFileno | int) -> None:
             obj = obj.fileno()
 
         if read_events.get(obj):
-            raise BusyResourceError("reading from") from None
+            raise BusyResourceError("reading from")
 
         loop = get_running_loop()
-        event = read_events[obj] = asyncio.Event()
-        loop.add_reader(obj, event.set)
+        event = asyncio.Event()
+        try:
+            loop.add_reader(obj, event.set)
+            remove_reader = loop.remove_reader
```

Comment:
Nit: I think this should be in the except's else clause

Reply:
Fixed.